openeogeotrellis/backend.py

```diff
@@ -2311,6 +2311,16 @@ def as_arg_element(dependency: dict) -> dict:
                 extra_py_files = ",".join(options.udf_dependency_files)
 
             job_work_dir = self.get_job_work_dir(job_id=job_id)
+            # Create the job work dir and set permissions.
+            ensure_dir(job_work_dir)
+            try:
+                shutil.chown(job_work_dir, user=None, group="openeo_results")
+            except LookupError as e:
+                logger.warning(f"Could not change group of {job_work_dir} to openeo_results.")
+            except PermissionError as e:
+                logger.warning(f"Could not change group of {job_work_dir} to openeo_results, no permissions.")
+            # Ensure others can read/write so that the batch job driver and executors can write to it.
+            add_permissions(job_work_dir, stat.S_IRWXO)
```

Comment:
wouldn't it be cleaner to do make the `shutil.chown` stuff an optional feature of `add_permissions`  (e.g if an optional `group` argument is given?

we don't need more spaghetti in `_start_job` I think.